lms/djangoapps/learner_dashboard/programs.py

```diff
@@ -220,7 +146,7 @@ def render_to_fragment(self, request, program_uuid, **kwargs):  # lint-amnesty,
             'industry_pathways': industry_pathways,
             'credit_pathways': credit_pathways,
             'program_discussions_enabled': program_discussions_is_enabled(),
-            'discussion_fragment': self.render_discussions_fragment(program_uuid, request)
+            'discussion_fragment': ProgramDiscussionLTI().render_discussions_iframe(program_uuid, request)
```

Comment:
there are some logical changes needed here. 
`ProgramDiscussionLTI().render_discussions_iframe(program_uuid, request)` do not return a "fragment" rather a data/context for program discussion.

Reply:
Here is what I would suggest. 

```
program_discussion = ProgramDiscussionLTI(program_uuid, request)

context = {
...
...
  'discussion' : {
      'enabled': program_discussion.enabled,
      'iframe': program_discussion.render_to_iframe()
  }
...
}

class ProgramDiscussionLTI:
    def __init__(self, program_uuid, request):
        self.program_uuid=program_uuid
        self.request = request
        self.enabled = program_discussions_is_enabled()
   ...
   ...
    def render_to_iframe(self):
        if not self.enabled:
            return ''
        program_discussions_configuration = self.get_program_discussion_configuration(program_uuid)
        if not program_discussions_configuration:
             return ''
       ...
       ...
       return fragment.content
```